lib/simulator-xcode-6.js

```diff
@@ -242,6 +242,52 @@ class SimulatorXcode6 {
     await simctl.eraseDevice(this.udid);
   }
 
+  async cleanCustomApp (appFile, appBundleId) {
+    log.debug(`Cleaning app data files`);
+
+    let appDirs = await this.getAppDirs(appFile, appBundleId);
+
+    if (appDirs.length === 0) {
+      log.debug("Couldn't find app directories to delete. Probably it's not " +
+               "installed");
+      return;
+    }
+
+    for (let dir of appDirs) {
+      log.debug(`Deleting directory: '${dir}'`);
+      await fs.rimraf(dir);
```

Comment:
These deletes can be done in parallel. See how it is done at lines 434-435 below.